### PR TITLE
add --ensure-channel flag for kots apps

### PR DESCRIPTION
**Pull request by dexhorthy**
Now you can do something like

```shell
replicated release create \
  --yaml-dir=./manifests/ \
  --promote=${GITHUB_BRANCH} \
  --ensure-channel
```

And it will promote the release to the `${GITHUB_BRANCH}` channel, creating that channel if it does not exist yet.

### CODE REVIEW

File: pkg/kotsclient/channel.go [hunk 1]
```diff
@@ -15,8 +15,14 @@ type GraphQLResponseGetChannel struct {
...
 }
+
+type GraphQLResponseCreateChannel struct {
+	Data   *KotsGetChannelData `json:"data,omitempty"`
+	Errors []graphql.GQLError  `json:"errors,omitempty"`
+}
+
 type KotsGetChannelData struct {
-	KotsChannel *KotsChannel `json:"getKotsChannel"`
+	KotsChannel *KotsChannel `json:"createKotsChannel"`
```
Comment (emosbaugh) [inline pkg/kotsclient/channel.go]: Weird naming

File: cli/cmd/runner.go [hunk 2]
```diff
@@ -41,27 +41,28 @@ type runnerArgs struct {
...
+	createReleaseDeploymentYaml       string
+	createReleaseServiceYaml          string
+	createReleasePreflightYaml        string
+	createReleaseSupportBundleYaml    string
+	createReleasePromote              string
+	createReleasePromoteDir           string
+	createReleasePromoteRequired      bool
+	createReleasePromoteNotes         string
+	createReleasePromoteVersion       string
+	createReleasePromoteEnsureChannel bool
```
Comment (dexhorthy) [inline cli/cmd/runner.go]: this is the new field

**Comment by dexhorthy** [comment]
Tested this locally against production, and tested an AKA application as well.

File: pkg/kotsclient/channel.go (hunk 1 above)
Comment (dexhorthy) [reply pkg/kotsclient/channel.go]: oh i might have goofed this, good catch

File: cli/cmd/release_create.go [hunk 3]
```diff
@@ -166,3 +163,42 @@ func (r *runners) releaseCreate(cmd *cobra.Command, args []string) error {
...
+		}
+	}
+
+	if len(promoteChannelIDs) == 0 {
+
+		// get-or-create only supported for KOTS apps at the moment
+		shouldCreateChannel := r.args.createReleasePromoteEnsureChannel && r.appType == "kots"
+
+		if shouldCreateChannel {
+			channelID, err := r.kotsAPI.CreateChannel(r.appID, r.args.createReleasePromote, "")
```
Comment (smithers1221) [inline cli/cmd/release_create.go]: is this arg createReleasePromote or createReleasePromoteDir for Kots app?

Comment (dexhorthy) [reply cli/cmd/release_create.go]: `createReleasePromote` is the *channel name* to promote the release to after creating -- I'd be open to updating the variable name in a followup PR if you think there's a way to make that more clear

**Comment by dexhorthy** [comment]
@smithers1221 I like the idea of messaging that to users though. will add

**Pull request merged**